index.js

```diff
@@ -8,29 +8,41 @@ const defaults = {
 	v4: '127.0.0.1'
 };
 
-function internalIp(family) {
-	return defaultGateway[family]().then(result => {
-		const interfaces = os.networkInterfaces();
-		const gatewayIp = ipaddr.parse(result.gateway);
-		let ret;
-
-		// Look for the matching interface in all local interfaces
-		Object.keys(interfaces).some(name => {
-			return interfaces[name].some(addr => {
-				const prefix = ipaddr.parse(addr.netmask).prefixLengthFromSubnetMask();
-				const net = ipaddr.parseCIDR(`${addr.address}/${prefix}`);
-
-				if (net[0].kind() === gatewayIp.kind() && gatewayIp.match(net)) {
-					ret = net[0].toString();
-				}
-
-				return Boolean(ret);
-			});
+function match(gateway, family) {
+	const interfaces = os.networkInterfaces();
+	const gatewayIp = ipaddr.parse(gateway);
+	let ret;
+
+	// Look for the matching interface in all local interfaces
+	Object.keys(interfaces).some(name => {
+		return interfaces[name].some(addr => {
+			const prefix = ipaddr.parse(addr.netmask).prefixLengthFromSubnetMask();
+			const net = ipaddr.parseCIDR(`${addr.address}/${prefix}`);
+
+			if (net[0].kind() === gatewayIp.kind() && gatewayIp.match(net)) {
```

Comment:
just noticed this: could you add a check for `net[0]` here, e.g. `if (net[0] && net[0].kind()`?

Reply:
this is a direct copy paste from the code as it presently is in master. mentioning only to confirm that you want to piggy back on this PR for the change

Reply:
right, I can fix that later myself, no need to do it here.